### PR TITLE
Add literal types to inferrer

### PR DESCRIPTION
**Pull request by vinistock**
### Motivation

This PR starts returning the expected type for literals, so that we can provide completion, go to definition, hover and signature help when something is invoked directly on one.

### Implementation

I also fixed a minor bug that we had not caught. We were showing any type in the completion documentation as if it were a guessed type, but that should only appear if it was guessed.

Then I added a bunch of branches to cover literals.

### Automated Tests

Added tests.

### CODE REVIEW

**Comment by andyw8** [comment]
Do we have a list of others literals not yet covered here? `Range` is the one I noticed, but there are probably others.

File: lib/ruby_lsp/type_inferrer.rb [hunk 1]
```diff
@@ -39,6 +39,26 @@ def infer_receiver_for_call_node(node, node_context)
       case receiver
       when Prism::SelfNode, nil
         self_receiver_handling(node_context)
+      when Prism::StringNode
```
Comment (Morriar) [inline lib/ruby_lsp/type_inferrer.rb:42]: I think you can also handle ranges and maybe procs

Comment (andyw8) [reply lib/ruby_lsp/type_inferrer.rb:42]: Also percent-literals (`%w{}` etc)?

Comment (vinistock) [reply lib/ruby_lsp/type_inferrer.rb:56]: Added ranges and literal lambdas. `%{}` are string nodes, so they are already handled.

**Pull request merged**